### PR TITLE
ignition.dsl: colcon builds for ign-transport

### PR DESCRIPTION
**Pull request by scpeters**
With the new dependency on ign-utils in https://github.com/ignitionrobotics/ign-transport/pull/216, which also uses colcon for its Windows CI builds, this may be a good time for ign-transport to use colcon as well. Since ign-transport4 is still supported, this requires https://github.com/ignition-tooling/gazebodistro/pull/29 before merging.

An alternative is to add a `configure.bat` script to `ign-utils` (https://github.com/ignitionrobotics/ign-utils/pull/13).

### CODE REVIEW

**Comment by j-rivero** [comment]
I've added 34c1058 my branch `colcon_ign_transport` with some more changes. Run a quick test here, but more branches need to be tested to be sure we are not breaking things: [Build Status](https://build.osrfoundation.org/job/_test_colcon_ignition_transport-ci-main-windows7-amd64/2/)

File: jenkins-scripts/dsl/ignition.dsl [hunk 1]
```diff
@@ -80,6 +80,7 @@ ignition_colcon_win         = [ 'gazebo',
                                 'physics',
                                 'rendering',
                                 'sensors',
+                                'transport',
```
Comment (j-rivero) [inline jenkins-scripts/dsl/ignition.dsl:83]: Adding transport here is the right thing, but name is going to change to get the shorter version. Just FYI.

File: jenkins-scripts/ign_transport-default-devel-windows-amd64.bat [hunk 2]
```diff
@@ -6,9 +6,8 @@ set VCS_DIRECTORY=ign-transport
 set PLATFORM_TO_BUILD=amd64
 set IGN_CLEAN_WORKSPACE=true
 
-for /f %%i in ('python "%SCRIPT_DIR%\tools\detect_cmake_major_version.py" "%WORKSPACE%\%VCS_DIRECTORY%\CMakeLists.txt"') do set IGN_TRANSPORT_MAJOR_VERSION=%%i
-if %IGN_TRANSPORT_MAJOR_VERSION% GEQ 4 (
-  call "%SCRIPT_DIR%/lib/generic-default-devel-windows.bat"
-) else (
-  call "%SCRIPT_DIR%/lib/ign_transport-base-windows.bat"
-)
+set DEPEN_PKGS="dlfcn-win32 cppzmq sqlite3 protobuf tinyxml2 zeromq"
```
Comment (scpeters) [inline jenkins-scripts/ign_transport-default-devel-windows-amd64.bat:9]: nit: alphabetize

File: jenkins-scripts/dsl/ignition.dsl (hunk 1 above)
Comment (scpeters) [reply jenkins-scripts/dsl/ignition.dsl:83]: ok, we'll need to update the ignition_collection dsl; I'll do that in a follow-up

File: jenkins-scripts/ign_transport-default-devel-windows-amd64.bat (hunk 2 above)
Comment (scpeters) [reply jenkins-scripts/ign_transport-default-devel-windows-amd64.bat]: f27d62f

**Comment by scpeters** [comment]
@j-rivero your changes look good, but I can't approve this since I submitted it

**Review by j-rivero — APPROVED** [review]

**Pull request merged**